src/zenml/integrations/pytorch/materializers/pytorch_materializer.py

```diff
@@ -49,4 +52,21 @@ def handle_return(self, model: Union[Module, TorchDict]) -> None:
             model: A torch.nn.Module or a dict to pass into model.save
         """
         super().handle_return(model)
-        torch.save(model, os.path.join(self.artifact.uri, DEFAULT_FILENAME))
+
+        # Save entire model to artifact directory, This is the default behavior for loading model in development phase (training, evaluation)
+        with fileio.open(
+            os.path.join(self.artifact.uri, DEFAULT_FILENAME), "wb"
+        ) as f:
+            torch.save(model, f)
+
+        # Save model checkpoint to artifact directory, This is the default behavior for loading model in production phase (inference)
+        if isinstance(model, Module):
+
+            with fileio.open(
+                os.path.join(self.artifact.uri, CHECKPOINT_FILENAME), "wb"
+            ) as f:
+                torch.save(model.state_dict(), f)
+
+        else:
```

Comment:
If we don't need to do anything here, could we perhaps remove the last two lines?

Reply:
yes sure, it's removed!